### PR TITLE
PYTHON-3127 Snapshot Query Examples for the Manual

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: test/test_examples.py [hunk 1]
```diff
@@ -1361,5 +1364,24 @@ def test_snapshot_query(self):
...
+# Wait for snapshot reads to become available to prevent this error:
+# [246:SnapshotUnavailable]: Unable to read from a snapshot due to pending collection catalog changes; please retry the operation. Snapshot timestamp is Timestamp(1646666892, 4). Collection minimum is Timestamp(1646666892, 5) (on localhost:27017, modern retry, attempt 1)
+# From https://github.com/mongodb/mongo-ruby-driver/commit/7c4117b58e3d12e237f7536f7521e18fc15f79ac
+def wait_for_snapshot(collection, client):
+    start_time = time.monotonic()
+
+    with client.start_session(snapshot=True) as s:
+        while True:
+            try:
+                collection.aggregate([{"$match": {"any": True}}], session=s)
```
Comment (ShaneHarvey) [inline test/test_examples.py:1376]: We should use a with statement to ensure the cursor is always closed. Also is the `{"$match":` needed or can we just do `aggregate([])`?

File: test/test_examples.py [hunk 2]
```diff
@@ -1361,5 +1364,24 @@ def test_snapshot_query(self):
         # End Snapshot Query Example 2
 
 
+# Wait for snapshot reads to become available to prevent this error:
+# [246:SnapshotUnavailable]: Unable to read from a snapshot due to pending collection catalog changes; please retry the operation. Snapshot timestamp is Timestamp(1646666892, 4). Collection minimum is Timestamp(1646666892, 5) (on localhost:27017, modern retry, attempt 1)
+# From https://github.com/mongodb/mongo-ruby-driver/commit/7c4117b58e3d12e237f7536f7521e18fc15f79ac
+def wait_for_snapshot(collection, client):
+    start_time = time.monotonic()
```
Comment (ShaneHarvey) [inline test/test_examples.py:1371]: This method can (should?) use the test/utils.py wait_until() method to retry.

File: test/test_examples.py [hunk 3]
```diff
@@ -1297,5 +1298,90 @@ def strptime(s):
         # End Versioned API Example 8
 
 
+class TestSnapshotQueryExamples(IntegrationTest):
+    @client_context.require_replica_set
```
Comment (ShaneHarvey) [inline test/test_examples.py:1302]: Why doesn't this test work on sharded clusters?

Comment (blink1073) [reply test/test_examples.py:1302]: From this [failure](https://spruce.mongodb.com/task/mongo_python_driver_tests_python_version_ubuntu18_test_ssl__platform~ubuntu_18.04_auth_ssl~auth_ssl_python_version~3.10_coverage~coverage_test_5.0_standalone_patch_861d79537fee9dd80be02a4ffdbadc5897acfd7d_623dd29ca4cf47697943fc09_22_03_25_14_33_01/tests?execution=0&sortBy=STATUS&sortDir=ASC):

`pymongo.errors.OperationFailure: node needs to be a replica set member to use readConcern: snapshot, full error: {'ok': 0.0, 'errmsg': 'node needs to be a replica set member to use readConcern: snapshot', 'code': 123, 'codeName': 'NotAReplicaSet'}`

File: test/test_examples.py (hunk 2 above)
Comment (blink1073) [reply test/test_examples.py:1371]: Done

File: test/test_examples.py (hunk 1 above)
Comment (blink1073) [reply test/test_examples.py:1376]: Done and Done

File: test/test_examples.py (hunk 3 above)
Comment (blink1073) [reply test/test_examples.py:1301]: Oh I see, either one should work

Comment (blink1073) [reply test/test_examples.py]: Updated

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**